polaris-react/src/types.ts

```diff
@@ -96,7 +96,7 @@ export interface BaseButton {
   /** Callback when element is touched */
   onTouchStart?(): void;
   /** Callback when pointerdown event is being triggered */
-  onPointerDown?(): void;
+  onPointerDown?(arg0?: any): void;
```

Comment:
is this cool for typescript? added it because above using the `Button` component and passing a function with an argument to `onPointerDown`

Reply:
It is fine to update the typing here, but it would be better if we could improve the naming and not use `any`.

Maybe try using something like `MouseEvent<'button'>`? I'm not sure of the exact types but can help jam on this if TS is giving you a lot of push back

Reply:
Let's hold off on merging this until we have discussed the larger implications of changing this typing. @aaronccasanova made a great point that changing this to `any` will impact our users and could make it difficult for us in the future reversing or changing this decision